xdsl/dialects/riscv_cf.py

```diff
@@ -240,6 +270,10 @@ class BltuOp(ConditionalBranchOperation):
 
     name = "riscv_cf.bltu"
 
+    def const_evaluate(self, rs1: int, rs2: int) -> bool:
+        # TODO: use proper bitwidth here?
+        return to_unsigned(rs1, 32) < to_unsigned(rs2, 32)
```

Comment:
This is what makes this complicated, we need to add an xlen to the rewrite, meaning also to the canonicalization infrastructure.

Reply:
TBH I think the rewritePattern base class should have an indexwidth property

Reply:
Interesting thought, let's put an issue up for this?

Reply:
#2060